File: apps/home-service-platform-server/src/APIs/Review/Dtos/ReviewCreateInput.cs
namespace HomeServicePlatform.APIs.Dtos;

public class ReviewCreateInput
{
    public string? Comment { get; set; }

    public DateTime CreatedAt { get; set; }

    public string? Id { get; set; }

    public int? Rating { get; set; }

    public ServiceProvider? ServiceProvider { get; set; }

    public List<ServiceProvider>? ServiceProviders { get; set; }

    public DateTime UpdatedAt { get; set; }

    public User? User { get; set; }

    public List<User>? Users { get; set; }
}

File: apps/home-service-platform-node-js-admin/src/api/review/ReviewCreateInput.ts
import { AdminCreateNestedManyWithoutReviewsInput } from "./AdminCreateNestedManyWithoutReviewsInput";
import { ServiceProviderWhereUniqueInput } from "../serviceProvider/ServiceProviderWhereUniqueInput";
import { ServiceProviderCreateNestedManyWithoutReviewsInput } from "./ServiceProviderCreateNestedManyWithoutReviewsInput";
import { UserWhereUniqueInput } from "../user/UserWhereUniqueInput";

export type ReviewCreateInput = {
  admins?: AdminCreateNestedManyWithoutReviewsInput;
  comment?: string | null;
  rating?: number | null;
  serviceProvider?: ServiceProviderWhereUniqueInput | null;
  serviceProviders?: ServiceProviderCreateNestedManyWithoutReviewsInput;
  user?: UserWhereUniqueInput | null;
};
